examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c

```diff
@@ -0,0 +1,813 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "silabs_utils.h"
+#include "sl_status.h"
+
+#include "FreeRTOS.h"
+#include "event_groups.h"
+#include "sl_board_configuration.h"
+#include "sl_net.h"
+#include "sl_si91x_host_interface.h"
+#include "sl_si91x_types.h"
+#include "sl_wifi_callback_framework.h"
+#include "sl_wifi_constants.h"
+#include "sl_wifi_types.h"
+#include "sl_wlan_config.h"
+#include "task.h"
+#include "wfx_host_events.h"
+
+#include "ble_config.h"
+
+#include "dhcp_client.h"
+#include "lwip/nd6.h"
+#include "sl_wifi.h"
+#include "wfx_host_events.h"
+#include "wfx_rsi.h"
+#define ADV_SCAN_THRESHOLD -40
+#define ADV_RSSI_TOLERANCE_THRESHOLD 5
+#define ADV_ACTIVE_SCAN_DURATION 15
+#define ADV_PASSIVE_SCAN_DURATION 20
+#define ADV_MULTIPROBE 1
+#define ADV_SCAN_PERIODICITY 10
+
+struct wfx_rsi wfx_rsi;
+
+/* Declare a variable to hold the data associated with the created event group. */
+StaticEventGroup_t rsiDriverEventGroup;
+
+bool hasNotifiedIPV6 = false;
+#if (CHIP_DEVICE_CONFIG_ENABLE_IPV4)
+bool hasNotifiedIPV4 = false;
+#endif /* CHIP_DEVICE_CONFIG_ENABLE_IPV4 */
+bool hasNotifiedWifiConnectivity = false;
+
+/* Declare a flag to differentiate between after boot-up first IP connection or reconnection */
+bool is_wifi_disconnection_event = false;
+
+/* Declare a variable to hold connection time intervals */
+uint32_t retryInterval              = WLAN_MIN_RETRY_TIMER_MS;
+volatile bool scan_results_complete = false;
+#define WIFI_SCAN_TIMEOUT 10000
```

Comment:
No comments on this timeout....

Reply:
Removed this file from this PR, adding it in a new PR